src/apps/activity_assignments/crud/assignments.py

```diff
@@ -23,3 +27,14 @@ async def already_exists(self, schema: ActivityAssigmentSchema) -> bool:
         query = query.exists()
         db_result = await self._execute(select(query))
         return db_result.scalars().first() or False
+
+    async def get_all_by_applet(self, applet_id: uuid.UUID) -> list[ActivityAssigmentSchema]:
+        query = (
+            select(ActivityAssigmentSchema)
+            .outerjoin(ActivitySchema, ActivitySchema.id == ActivityAssigmentSchema.activity_id)
+            .outerjoin(ActivityFlowSchema, ActivityFlowSchema.id == ActivityAssigmentSchema.activity_flow_id)
+            .where(or_(ActivityFlowSchema.applet_id == applet_id, ActivitySchema.applet_id == applet_id))
```

Comment:
I've discovered that assignments are still returned **even after the activity, respondent subject, or target subject are deleted** from the applet. What is the expected behaviour for this:
- Should soft-deleting the activity/flow, respondent subject, or target subject also soft-delete any assignments associated with those entities?
- Or should this query just be updated to filter out any assignments associated with soft-deleted entities?

I feel that the current behaviour, where all records associated with deleted entities are still being returned by this query, isn't correct. What are your thoughts?

Reply:
There is no soft deleted activity or flow, all are hard deleted during update for an applet

Reply:
So, with your latest changes, I've noticed the following two behaviours:

- When I **delete an activity or activity flow**, the endpoint no longer returns assignments associated with that activity or activity flow. This is the desired functionality from the FE's point of view – thank you for that. However, I checked the `activity_assignments` table, and those assignments that are associated with now-deleted activities/activity flows still exist, and `is_deleted` is still set to `FALSE`. Shouldn't those assignments be soft-deleted when the associated activity or activity flow has been deleted?
- When I **delete a subject**, the endpoint still returns assignments associated with that deleted subject (whether it's the `respondent_subject_id` or `target_subject_id`). I don't think the FE should continue seeing assignments associated with now-deleted subjects. Furthermore, I think the assignments should also be soft-deleted in the `activity_assignments` table (same as with deleted activities/flows). What do you think?